fapi-client/src/main/scala/com/gu/facia/api/Response.scala

```diff
@@ -15,6 +15,13 @@ case class Response[A] protected (underlying: Future[Either[ApiError, A]]) {
     }
   }
 
+  def filter(f: A => Boolean, orLeft: ApiError = FilteredOut)(implicit ex: ExecutionContext): Response[A] = flatMap { a =>
```

Comment:
Yeah agreed. I suggested pointing `withFilter` to `filter` for the time being until you figure out how to make it truly lazy.